%pip install pyodbc  
%pip install python-dotenv 

Hypothesis

QUESTIONS
1. Which companies received the highest amount of funding? 
2. At what funding stage did investors fund the highest?
3. Which sectors/industry's received the highest amount of funding? 
4. Which sectors/industry's received the least amount of funding? 
5. Which cities received the highest amount of funding ?
6. Which companies received the least amount of funding?



 

In [41]:
pip install scikit-learn


     ---------------------------------------- 9.2/9.2 MB 310.8 kB/s eta 0:00:00
     -------------------------------------- 44.0/44.0 MB 790.9 kB/s eta 0:00:00
     ------------------------------------ 302.0/302.0 kB 444.7 kB/s eta 0:00:00



[notice] A new release of pip available: 22.3.1 -> 23.1.2
[notice] To update, run: python.exe -m pip install --upgrade pip


In [42]:
from sklearn.impute import SimpleImputer
import numpy as np

In [2]:
#  a package for creating connection strings to your remote database
%pip install pyodbc

Note: you may need to restart the kernel to use updated packages.



[notice] A new release of pip available: 22.3.1 -> 23.1.2
[notice] To update, run: python.exe -m pip install --upgrade pip


In [3]:
# a package for creating environment variables that will help you hide sensitve configuration information such as database credentials and API keys
%pip install python-dotenv


Note: you may need to restart the kernel to use updated packages.



[notice] A new release of pip available: 22.3.1 -> 23.1.2
[notice] To update, run: python.exe -m pip install --upgrade pip


import pyodbc #just installed with pip
from dotenv import dotenv_values #import the dotenv_values function from the dotenv package
import pandas as pd
import warnings 

warnings.filterwarnings('ignore')

In [4]:
import pyodbc
from dotenv import dotenv_values
import pandas as pd
import warnings
warnings.filterwarnings('ignore')

In [53]:
# Load environment variables from .env file into a dictionary
environment_variables = dotenv_values('.env')


# Get the values for the credentials you set in the '.env' file
database = environment_variables.get("DATABASE")
server = environment_variables.get("SERVER")
username = environment_variables.get("USERNAME")
password = environment_variables.get("PASSWORD")

connection_string = f"DRIVER={{SQL Server}};SERVER={server};DATABASE={database};UID={username};PWD={password}"

In [54]:
# Use the connect method of the pyodbc library and pass in the connection string.
# This will connect to the server and might take a few seconds to be complete. 
# Check your internet connection if it takes more time than necessary

connection = pyodbc.connect(connection_string)

In [7]:
#2018 DATASET
data_2018 = pd.read_csv("startup_funding2018.csv")
data_2018.head()

Company Name                                           Industry   
0  TheCollegeFever  Brand Marketing, Event Promotion, Marketing, S...  \
1  Happy Cow Dairy                               Agriculture, Farming   
2       MyLoanCare   Credit, Financial Services, Lending, Marketplace   
3      PayMe India                        Financial Services, FinTech   
4         Eunimart                 E-Commerce Platforms, Retail, SaaS   

  Round/Series       Amount                          Location   
0         Seed       250000       Bangalore, Karnataka, India  \
1         Seed  ₹40,000,000        Mumbai, Maharashtra, India   
2     Series A  ₹65,000,000           Gurgaon, Haryana, India   
3        Angel      2000000       Noida, Uttar Pradesh, India   
4         Seed            —  Hyderabad, Andhra Pradesh, India   

                                       About Company  
0  TheCollegeFever is a hub for fun, fiesta and f...  
1  A startup which aggregates milk from dairy far...  
2          Leading Online Loans Marketplace in India  
3  PayMe India is an innovative FinTech organizat...  
4  Eunimart is a one stop solution for merchants ...

In [8]:
data_2018.shape

(526, 6)

In [9]:
data_2018.columns

Index(['Company Name', 'Industry', 'Round/Series', 'Amount', 'Location',
       'About Company'],
      dtype='object')

In [10]:
data_2018.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 526 entries, 0 to 525
Data columns (total 6 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   Company Name   526 non-null    object
 1   Industry       526 non-null    object
 2   Round/Series   526 non-null    object
 3   Amount         526 non-null    object
 4   Location       526 non-null    object
 5   About Company  526 non-null    object
dtypes: object(6)
memory usage: 24.8+ KB


In [11]:
data_2018.rename(columns={
    'Company Name':'Company_Brand',
    'Industry':'Sector',
    'About Company':'What_it_does',
    'Amount':'Amount($)',
    'Location':'HeadQuarter',
    'Round/Series':'Stage',
}, inplace=True)

data_2018.head()


Company_Brand                                             Sector   
0  TheCollegeFever  Brand Marketing, Event Promotion, Marketing, S...  \
1  Happy Cow Dairy                               Agriculture, Farming   
2       MyLoanCare   Credit, Financial Services, Lending, Marketplace   
3      PayMe India                        Financial Services, FinTech   
4         Eunimart                 E-Commerce Platforms, Retail, SaaS   

      Stage    Amount($)                       HeadQuarter   
0      Seed       250000       Bangalore, Karnataka, India  \
1      Seed  ₹40,000,000        Mumbai, Maharashtra, India   
2  Series A  ₹65,000,000           Gurgaon, Haryana, India   
3     Angel      2000000       Noida, Uttar Pradesh, India   
4      Seed            —  Hyderabad, Andhra Pradesh, India   

                                        What_it_does  
0  TheCollegeFever is a hub for fun, fiesta and f...  
1  A startup which aggregates milk from dairy far...  
2          Leading Online Loans Marketplace in India  
3  PayMe India is an innovative FinTech organizat...  
4  Eunimart is a one stop solution for merchants ...

In [12]:
# data_2018['Amount'].unique

In [52]:
#change the amount datatype
# data_2018['Amount($)'] = data_2018['Amount($)'].astype(int)

In [18]:
data_2018.duplicated().sum()

1

In [19]:
data_2018.drop_duplicates()



Company_Brand                                             Sector   
0      TheCollegeFever  Brand Marketing, Event Promotion, Marketing, S...  \
1      Happy Cow Dairy                               Agriculture, Farming   
2           MyLoanCare   Credit, Financial Services, Lending, Marketplace   
3          PayMe India                        Financial Services, FinTech   
4             Eunimart                 E-Commerce Platforms, Retail, SaaS   
..                 ...                                                ...   
521              Udaan   B2B, Business Development, Internet, Marketplace   
522  Happyeasygo Group                                    Tourism, Travel   
523             Mombay         Food and Beverage, Food Delivery, Internet   
524         Droni Tech                             Information Technology   
525            Netmeds         Biotechnology, Health Care, Pharmaceutical   

        Stage    Amount($)                       HeadQuarter   
0        Seed       250000       Bangalore, Karnataka, India  \
1        Seed  ₹40,000,000        Mumbai, Maharashtra, India   
2    Series A  ₹65,000,000           Gurgaon, Haryana, India   
3       Angel      2000000       Noida, Uttar Pradesh, India   
4        Seed            —  Hyderabad, Andhra Pradesh, India   
..        ...          ...                               ...   
521  Series C    225000000       Bangalore, Karnataka, India   
522  Series A            —           Haryana, Haryana, India   
523      Seed         7500        Mumbai, Maharashtra, India   
524      Seed  ₹35,000,000        Mumbai, Maharashtra, India   
525  Series C     35000000        Chennai, Tamil Nadu, India   

                                          What_it_does  
0    TheCollegeFever is a hub for fun, fiesta and f...  
1    A startup which aggregates milk from dairy far...  
2            Leading Online Loans Marketplace in India  
3    PayMe India is an innovative FinTech organizat...  
4    Eunimart is a one stop solution for merchants ...  
..                                                 ...  
521  Udaan is a B2B trade platform, designed specif...  
522            HappyEasyGo is an online travel domain.  
523  Mombay is a unique opportunity for housewives ...  
524  Droni Tech manufacture UAVs and develop softwa...  
525       Welcome to India's most convenient pharmacy!  

[525 rows x 6 columns]

In [20]:
#clean Amount column
data_2018['Amount($)']

0           250000
1      ₹40,000,000
2      ₹65,000,000
3          2000000
4                —
          ...     
521      225000000
522              —
523           7500
524    ₹35,000,000
525       35000000
Name: Amount($), Length: 526, dtype: object

In [21]:
data_2018.isnull().sum()

Company_Brand    0
Sector           0
Stage            0
Amount($)        0
HeadQuarter      0
What_it_does     0
dtype: int64

Summary for 2018 dataset
The shape for the 2018 data is 526 rows and 6 columns
No missing values
Colunm renamed to name related to the project
One duplicated value
/////////////////////////////////////////////
Amount column need to be cleaned
Amount column need to be changed to int or float from object

In [22]:
#2019 data
data_2019 = pd.read_csv("startup_funding2019.csv")

In [23]:
# data_2018.head()

In [24]:
data_2019.head()

Company/Brand  Founded HeadQuarter           Sector   
0  Bombay Shaving      NaN         NaN        Ecommerce  \
1       Ruangguru   2014.0      Mumbai           Edtech   
2        Eduisfun      NaN      Mumbai           Edtech   
3        HomeLane   2014.0     Chennai  Interior design   
4        Nu Genes   2004.0   Telangana         AgriTech   

                                        What it does   
0         Provides a range of male grooming products  \
1  A learning platform that provides topic-based ...   
2            It aims to make learning fun via games.   
3              Provides interior designing solutions   
4  It is a seed company engaged in production, pr...   

                                Founders   
0                     Shantanu Deshpande  \
1  Adamas Belva Syah Devara, Iman Usman.   
2                          Jatin Solanki   
3           Srikanth Iyer, Rama Harinath   
4                 Narayana Reddy Punyala   

                                            Investor     Amount($)   
0                               Sixth Sense Ventures    $6,300,000  \
1                                   General Atlantic  $150,000,000   
2     Deepak Parekh, Amitabh Bachchan, Piyush Pandey   $28,000,000   
3  Evolvence India Fund (EIF), Pidilite Group, FJ...   $30,000,000   
4           Innovation in Food and Agriculture (IFA)    $6,000,000   

           Stage  
0            NaN  
1       Series C  
2  Fresh funding  
3       Series D  
4            NaN

In [25]:
data_2019.shape

(89, 9)

In [27]:
# data_2018.columns

Index(['Company_Brand', 'Sector', 'Stage', 'Amount($)', 'HeadQuarter',
       'What_it_does'],
      dtype='object')

In [26]:
data_2019.columns

Index(['Company/Brand', 'Founded', 'HeadQuarter', 'Sector', 'What it does',
       'Founders', 'Investor', 'Amount($)', 'Stage'],
      dtype='object')

In [31]:
column_to_drop_2019 = ["Founded","Founders","Investor"]
data_2019 = data_2019.drop(column_to_drop_2019, axis=1)



In [32]:
data_2019.columns

Index(['Company/Brand', 'HeadQuarter', 'Sector', 'What it does', 'Amount($)',
       'Stage'],
      dtype='object')

In [33]:
data_2019.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 89 entries, 0 to 88
Data columns (total 6 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   Company/Brand  89 non-null     object
 1   HeadQuarter    70 non-null     object
 2   Sector         84 non-null     object
 3   What it does   89 non-null     object
 4   Amount($)      89 non-null     object
 5   Stage          43 non-null     object
dtypes: object(6)
memory usage: 4.3+ KB


In [36]:
# missing values
data_2019.isnull().sum()

Company/Brand     0
HeadQuarter      19
Sector            5
What it does      0
Amount($)         0
Stage            46
dtype: int64

In [47]:
data_2019['HeadQuarter']

0           NaN
1        Mumbai
2        Mumbai
3       Chennai
4     Telangana
        ...    
84       Mumbai
85     Gurugram
86        Delhi
87        Delhi
88    Rajasthan
Name: HeadQuarter, Length: 89, dtype: object

In [46]:
# imputer = SimpleImputer(strategy='mean')
# X = ["HeadQuarter"]
# imputer.fit(X)
# X_imputed = imputer.transform(X)


2020 DATASET $ CLEANING

In [55]:
query = "Select * from dbo.LP1_startup_funding2020"
data_2020 = pd.read_sql(query, connection)

In [77]:
# data_2020.head(50)
data_2020.tail(50)

Company_Brand HeadQuarter                  Sector   
1005           Extra Edge        Pune                  Edtech  \
1006     Capital Quotient   Bangalore                 Fintech   
1007                Karfa       Delhi               Ecommerce   
1008              Gamerji       Delhi                  Gaming   
1009               Bounce   Bangalore  Automotive and Rentals   
1010               Zinier   Bangalore              Automation   
1011              ProEves       Delhi       Preschool Daycare   
1012               Toddle   Bengaluru                    None   
1013  Aesthetic Nutrition   Bengaluru               Food diet   
1014        Capital Float   Bangalore                 Fintech   
1015            Ninjacart   Bangalore            B2B Agritech   
1016          TechnifyBiz   Bangalore            B2B Agritech   
1017                 Wysh   Hyderabad               E-connect   
1018                Keito        Pune                      AI   
1019                 Klub        Pune                 Fintech   
1020               Aadhar      Mumbai               Ecommerce   
1021          IncubateHub   Bengaluru                Tech hub   
1022          Rage Coffee       Delhi                    FMCG   
1023            Skilancer       Noida              Technology   
1024    Harappa Education   New Delhi                  Edtech   
1025                 Rheo   New Delhi           Entertainment   
1026    Axio Biosolutions   Bangalore                 Medtech   
1027            Techbooze   Bangalore             Consultancy   
1028              PayMart  Chandigarh                 Fintech   
1029             Testbook  Chandigarh                  Edtech   
1030             FabAlley       Noida               Ecommerce   
1031             HaikuJAM       Noida                    None   
1032            Mamaearth       Noida               Ecommerce   
1033            Rentomojo       Noida                  Rental   
1034               Zomato       Noida                Foodtech   
1035                 Byju       Noida                  Edtech   
1036             Generico      Mumbai                Pharmacy   
1037           HighRadius   Hyderabad                    SaaS   
1038           Chai Kings     Chennai                Beverage   
1039               Rivigo    Gurugram               Logistics   
1040             BuildPan      Indore                    SaaS   
1041              Shiksha      Indore                  Edtech   
1042       Super Surfaces   Hyderabad               Deisgning   
1043               Quicko   Ahmedabad                Taxation   
1044     Satin Creditcare     Gurgaon                 Fintech   
1045           XpressBees     Gurgaon               Logistics   
1046              FarmERP        Pune                Agritech   
1047        Wealth Bucket        Pune                 Fintech   
1048           GoMechanic       Delhi   Automobile Technology   
1049               Fashor     Chennai                 Fashion   
1050         Leverage Edu       Delhi                  Edtech   
1051                EpiFi       Delhi                 Fintech   
1052              Purplle      Mumbai               Cosmetics   
1053               Shuttl       Delhi               Transport   
1054                Pando     Chennai                Logitech   

                                           What_it_does   
1005  Admission Marketing Automation software for th...  \
1006                                 Investment advisor   
1007              It sells products crafted by artisans   
1008                        Esports tournament platform   
1009  Offers a variety of bikes and scooters that ca...   
1010      Provides intelligent field service automation   
1011  Provides tech-enabled, end-to-end childcare su...   
1012  Collaboration and productivity platform for te...   
1013     Markets and sells chewable dietary supplements   
1014                                    Digital lending   
1015                   Online grocery delivery 

In [57]:
data_2020.shape

(1055, 10)

In [58]:
data_2020.columns

Index(['Company_Brand', 'Founded', 'HeadQuarter', 'Sector', 'What_it_does',
       'Founders', 'Investor', 'Amount', 'Stage', 'column10'],
      dtype='object')

In [59]:
data_2020.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1055 entries, 0 to 1054
Data columns (total 10 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Company_Brand  1055 non-null   object 
 1   Founded        842 non-null    float64
 2   HeadQuarter    961 non-null    object 
 3   Sector         1042 non-null   object 
 4   What_it_does   1055 non-null   object 
 5   Founders       1043 non-null   object 
 6   Investor       1017 non-null   object 
 7   Amount         801 non-null    float64
 8   Stage          591 non-null    object 
 9   column10       2 non-null      object 
dtypes: float64(2), object(8)
memory usage: 82.5+ KB


In [60]:

#finding the missing value
data_2020.isnull().sum()

Company_Brand       0
Founded           213
HeadQuarter        94
Sector             13
What_it_does        0
Founders           12
Investor           38
Amount            254
Stage             464
column10         1053
dtype: int64

In [62]:
#Drop the columns that are not needed.
column_to_drop_2020 = ["Founded", "Founders", "column10"]
data_2020 = data_2020.drop(column_to_drop_2020, axis=1)

In [63]:
# check if the columns have been dropped
data_2020.columns

Index(['Company_Brand', 'HeadQuarter', 'Sector', 'What_it_does', 'Investor',
       'Amount', 'Stage'],
      dtype='object')

In [65]:
#Unwanted columns have been dropped
data_2020.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1055 entries, 0 to 1054
Data columns (total 7 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Company_Brand  1055 non-null   object 
 1   HeadQuarter    961 non-null    object 
 2   Sector         1042 non-null   object 
 3   What_it_does   1055 non-null   object 
 4   Investor       1017 non-null   object 
 5   Amount         801 non-null    float64
 6   Stage          591 non-null    object 
dtypes: float64(1), object(6)
memory usage: 57.8+ KB


In [66]:
#The company_brand colunm
data_2020["Company_Brand"].unique()

array(['Aqgromalin', 'Krayonnz', 'PadCare Labs', 'NCOME', 'Gramophone',
       'qZense', 'MyClassboard', 'Metvy', 'Rupeek', 'Gig India',
       'Slurrp Farm', 'Medfin', 'MasterG', 'Brila 91', 'FoodyBuddy',
       'Zomato', 'OurEye.ai', 'Shiprocket', 'Pine Labs',
       'Moneyboxx Finance', 'EWar', 'SucSEED Indovation', 'TaxBuddy.com',
       'Awign Enterprises', 'Circle of Angels', 'CoinDCX', 'Glance',
       'VerSe Innovation', 'Voosh', 'Delhivery', 'Arya', 'Pickrr',
       'PhonePe', 'Countingwell', 'Callify.ai', 'Hudle', 'Cover it Up',
       'BiteSpeed', 'WakeFit', 'Sreyas Holistic Remedies', 'Fleeca India',
       'M1xchange', 'Exotel', 'PagarBook', 'PointOne Capital', 'Ayana',
       'Fisdom', 'Tinker Coders', 'Infinite Uptime', 'HappyLocate',
       'Loop Health', 'NeoDocs', 'The ePlane', 'TagMango', 'The New Shop',
       'MobiKwik', 'DebugsBunny', 'BHIVE Investech', 'Rivigo', 'ZipLoan',
       'Swipez', 'Humus', 'Artivatic.ai', 'Prescinto', 'Toppeq',
       'Bolo Indya', 'Bomb

In [67]:
#Headquaters column
data_2020["HeadQuarter"].unique()


array(['Chennai', 'Bangalore', 'Pune', 'New Delhi', 'Indore', 'Hyderabad',
       'Gurgaon', 'Belgaum', 'Noida', 'Mumbai', 'Andheri', 'Jaipur',
       'Ahmedabad', 'Kolkata', 'Tirunelveli, Tamilnadu', 'Thane', None,
       'Singapore', 'Gurugram', 'Gujarat', 'Haryana', 'Kerala', 'Jodhpur',
       'Jaipur, Rajastan', 'Delhi', 'Frisco, Texas, United States',
       'California', 'Dhingsara, Haryana', 'New York, United States',
       'Patna', 'San Francisco, California, United States',
       'San Francisco, United States', 'San Ramon, California',
       'Paris, Ile-de-France, France', 'Plano, Texas, United States',
       'Sydney', 'San Francisco Bay Area, Silicon Valley, West Coast',
       'Bangaldesh', 'London, England, United Kingdom',
       'Sydney, New South Wales, Australia', 'Milano, Lombardia, Italy',
       'Palmwoods, Queensland, Australia', 'France',
       'San Francisco Bay Area, West Coast, Western US',
       'Trivandrum, Kerala, India', 'Cochin', 'Samastipur, Bihar',


In [ ]:
# Standardize city names
# df20["HeadQuarter"] = df20["HeadQuarter"].str.replace("Bangalore|Bengaluru", "Bangalore")
# df20["HeadQuarter"] = df20["HeadQuarter"].str.replace("Hyderebad", "Hyderabad")
# df20["HeadQuarter"] = df20["HeadQuarter"].str.replace("New Delhi|Delhi", "New Delhi")

In [72]:
# Check for missing values
data_2020["HeadQuarter"].isnull().sum()

94

In [73]:
#Handling missing values
# df20["HeadQuarter"] = df20["HeadQuarter"].replace(["", "-", "nan"], np.nan)
data_2020["HeadQuarter"]= data_2020["HeadQuarter"].replace(["",np.nan])

In [78]:
#No more missing values in Headquarter column
data_2020["HeadQuarter"].isnull().sum()

0

In [81]:
data_2020.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1055 entries, 0 to 1054
Data columns (total 7 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Company_Brand  1055 non-null   object 
 1   HeadQuarter    1055 non-null   object 
 2   Sector         1042 non-null   object 
 3   What_it_does   1055 non-null   object 
 4   Investor       1017 non-null   object 
 5   Amount         801 non-null    float64
 6   Stage          591 non-null    object 
dtypes: float64(1), object(6)
memory usage: 57.8+ KB


In [82]:
#The Sector Column
data_2020["Sector"].unique()

array(['AgriTech', 'EdTech', 'Hygiene management', 'Escrow',
       'Networking platform', 'FinTech', 'Crowdsourcing',
       'Food & Bevarages', 'HealthTech', 'Fashion startup',
       'Food Industry', 'Food Delivery', 'Virtual auditing startup',
       'E-commerce', 'Gaming', 'Work fulfillment', 'AI startup',
       'Telecommunication', 'Logistics', 'Tech Startup', 'Sports',
       'Retail', 'Medtech', 'Tyre management', 'Cloud company',
       'Software company', 'Venture capitalist', 'Renewable player',
       'IoT startup', 'SaaS startup', 'Aero company', 'Marketing company',
       'Retail startup', 'Co-working Startup', 'Finance company',
       'Tech company', 'Solar Monitoring Company',
       'Video sharing platform', 'Gaming startup',
       'Video streaming platform', 'Consumer appliances',
       'Blockchain startup', 'Conversational AI platform', 'Real Estate',
       'SaaS platform', 'AI platform', 'Fusion beverages', 'HR Tech',
       'Job portal', 'Hospitality', 'Digit

In [84]:
data_2020["Sector"].isnull().sum()

13

In [ ]:
query_2021 = "Select * from dbo.LP1_startup_funding2021"
data_2021 = pd.read_sql(query_2021, connection)

: 

In [ ]:
data_2021.head()

: 

In [ ]:
data_2021.shape

: 

Hypothesis
H0 = Funding is highly dependent on the industry/sector a company is in.
Ha = Funding is not dependent on industry/sector.



Questions
1. Which sector/industry recevied the highest amount?
2. At what stage/round can a sector/industry received the highest amount?
3. Which location have the highest amount?
4. Which location have the least amount?
5. What does each company do that make them to receive the highest amount?